### PR TITLE
Forms update with CollectionView conversion

### PR DESCRIPTION
**Pull request by mportune-bw**
**Overview**

- Updates to Xamarin Forms dependencies
- Conversion to CollectionView throughout app (from ListView)
- Additional UI tweaks

**Files Changed/Added**

Android:

- **CipherViewCellRenderer.cs**: DELETED, `ViewCell` not used with `CollectionView`
- **ExtendedGridRenderer.cs**: Required to add pressed-state color support to Grid views within `CollectionView`
- **ExtendedListViewRenderer.cs**: DELETED, `ListView` replaced with `CollectionView`
- **ExtendedStackLayoutRenderer.cs**: Required to add pressed-state color support to StackLayouts within `CollectionView`
- **SendViewCellRenderer.cs**: DELETED, `ViewCell` not used with `CollectionView`
- **list_item_bg.xml**: Background selector for pressed item state with light theme
- **list_item_bg_dark.xml**: Background selector for pressed item state with dark theme
- **list_item_bg_nord.xml**: Background selector for pressed item state with nord theme
- **CipherViewCell.axml**: DELETED, No longer used with `CollectionView`
- **SendViewCell.axml**: DELETED, No longer used with `CollectionView`
- **colors.xml**: Added `itemPressed` color for light theme (dark & nord use `primary` but that was too dark for light theme)
- **Android.csproj**: Updated dependencies

App:

- **CipherViewCell.xaml**: Converted to `ExtendedGrid` for proper `CollectionView` support
- **CipherViewCell.xaml.cs**: Converted to `ExtendedGrid` and adjusted code accordingly
- **CipherViewCellViewModel.cs**: Added constructor plus `WebsiteIconsEnabled` and `ShowIconImage` to support new implementation
- **SendViewCell.xaml**: Converted to `ExtendedGrid` for proper `CollectionView` support
- **SendViewCell.xaml.cs**: Converted to `ExtendedGrid` and adjusted code accordingly
- **SendViewCellViewModel.cs**: Added constructor to support new implementation
- **ExtendedCollectionView.cs**: `CollectionView` subclass for use with iOS pressed-state customizations (will be added later)
- **ExtendedGrid.cs**: `Grid` subclass for use with Android pressed-state customizations
- **ExtendedListView.cs**: DELETED, No longer used with `CollectionView`
- **ExtendedStackLayout.cs**: `StackLayout` subclass for use with Android pressed-state customizations
- **GeneratorHistoryPage.xaml**: Converted to `ExtendedCollectionView` where necessary
- **SendGroupingsPage.xaml**: Converted to `ExtendedStackLayout` and `ExtendedCollectionView` where necessary
- **SendGroupingsPage.xaml.cs**: Converted to `ExtendedCollectionView` and fixed issue with "About" icon blocking the back button in the toolbar in iOS, and missing ActivityIndicator on initial load
- **SendGroupingsPageListGroup.cs**: Renamed `groupItems` to `SendGroupItems` for consistency
- **SendGroupingsPageViewModel.cs**: Removed `ShowAddSendButton` as it's now handled elsewhere
- **SendAddEditPageViewModel.cs**: Moved share sheet trigger to happen after "back" function fires to prevent the sheet from disappearing in some iOS configurations
- **SendsPage.xaml**: Converted to `ExtendedCollectionView` where necessary
- **SendsPage.xaml.cs**: Updated `RowSelected` to support `ExtendedCollectionView`
- **SettingsPage.xaml**: Converted to `ExtendedStackLayout` and `ExtendedCollectionView` where necessary
- **SettingsPage.xaml.cs**: Updated `RowSelected` to support `ExtendedCollectionView`
- **FoldersPage.xaml**: Converted to `ExtendedCollectionView` where necessary
- **FoldersPage.xaml.cs**: Updated `RowSelected` to support `ExtendedCollectionView`
- **GroupingsPage.xaml**: Converted to `ExtendedStackLayout` and `ExtendedCollectionView` where necessary
- **GroupingsPage.xaml.cs**: Updated `RowSelected` to support `ExtendedCollectionView`
- **AutofillCiphersPage.xaml**: Converted to `ExtendedCollectionView` where necessary
- **AutofillCiphersPage.xaml.cs**: Updated `RowSelected` to support `ExtendedCollectionView`
- **CiphersPage.xaml**: Converted to `ExtendedCollectionView` where necessary
- **CiphersPage.xaml.cs**: Updated `RowSelected` to support `ExtendedCollectionView`
- **PasswordHistoryPage.xaml**: Converted to `ExtendedCollectionView` where necessary
- **AppResources.resx/Designer.cs**: Added `AddSend` string for no-longer-missing "Add Send" button shown with empty Send list
- **Android.xaml**: Added new `CollectionView` styles
- **Base.xaml**: Added new `CollectionView` styles
- **iOS.xaml**: Added new `CollectionView` styles
- **IconGlyphConverter.cs**: Converter for showing FontAwesome icons in Cipher CollectionView
- **IconImageConverter.cs**: Converter for showing favicon bitmaps in Cipher CollectionView
- **SendIconGlyphConverter.cs**: Converter for showing FontAwesome icons in Send CollectionView
- **App.csproj**: Updated dependencies

iOS:

- **iOS.csproj**: Updated dependencies

iOS.Core:

- **DeviceActionService.cs**: Removed check for visible toolbar when showing toast due to inconsistencies with timing, and adjusted margins accordingly to be padded regardless
- **Toast.cs**: Adjusted `CornerRadius` to match iOS widgets better

**Screenshots**

### CODE REVIEW

File: src/App/Controls/CipherViewCell/CipherViewCellViewModel.cs [hunk 1]
```diff
@@ -6,11 +6,29 @@ namespace Bit.App.Controls
...
+
+        public bool WebsiteIconsEnabled
+        {
+            get => _websiteIconsEnabled;
+            set => SetProperty(ref _websiteIconsEnabled, value);
+        }
+
+        public bool ShowIconImage
+        {
+            get => WebsiteIconsEnabled && Cipher.Login != null && !string.IsNullOrWhiteSpace(Cipher.Login.Uri);
```
Comment (cscharf) [inline src/App/Controls/CipherViewCell/CipherViewCellViewModel.cs:31]: This _could_ be simplified:
```suggestion
            get => WebsiteIconsEnabled && !string.IsNullOrWhiteSpace(Cipher.Login?.Uri);
```
```suggestion
            get => WebsiteIconsEnabled && Cipher.Login != null && !string.IsNullOrWhiteSpace(Cipher.Login.Uri);
```

File: src/App/Controls/CipherViewCell/CipherViewCell.xaml.cs [hunk 2]
```diff
@@ -1,45 +1,26 @@
...
+﻿using System;
 using Bit.Core.Models.View;
-using Bit.Core.Utilities;
-using System;
 using Xamarin.Forms;
 
 namespace Bit.App.Controls
 {
-    public partial class CipherViewCell : ViewCell
+    public partial class CipherViewCell : ExtendedGrid
```
Comment (cscharf) [inline src/App/Controls/CipherViewCell/CipherViewCell.xaml.cs:7]: Nice refactor!

File: src/Android/Renderers/ExtendedStackLayoutRenderer.cs [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+            if (_bgResId == null)
+            {
+                if (ThemeManager.GetTheme(true) == "nord")
+                {
+                    _bgResId = Resource.Drawable.list_item_bg_nord;
+                }
+                else
+                {
+                    _bgResId ??= ThemeManager.UsingLightTheme ? Resource.Drawable.list_item_bg : 
+                Resource.Drawable.list_item_bg_dark;
```
Comment (cscharf) [inline src/Android/Renderers/ExtendedStackLayoutRenderer.cs:37]: Some wonky formatting here on indentation.

File: src/Android/Renderers/ExtendedGridRenderer.cs [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+            if (_bgResId == null)
+            {
+                if (ThemeManager.GetTheme(true) == "nord")
+                {
+                    _bgResId = Resource.Drawable.list_item_bg_nord;
+                }
+                else
+                {
+                    _bgResId ??= ThemeManager.UsingLightTheme ? Resource.Drawable.list_item_bg : 
+                Resource.Drawable.list_item_bg_dark;
```
Comment (cscharf) [inline src/Android/Renderers/ExtendedGridRenderer.cs:37]: Issue here on indenting of new line

File: src/Android/Renderers/ExtendedGridRenderer.cs [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+            {
+                SetBackgroundResource(GetBgResId());
+            }
+        }
+
+        private int GetBgResId()
+        {
+            if (_bgResId == null)
+            {
+                if (ThemeManager.GetTheme(true) == "nord")
```
Comment (cscharf) [inline src/Android/Renderers/ExtendedGridRenderer.cs:30]: Is there a reason `"nord"` is a special case in the way the renderer handles theming?

**Review by cscharf — COMMENTED** [review]
Looks good overall, great refactor (I know this was a huge effort). Just some minor feedback and open questions.

File: src/App/Utilities/IconGlyphConverter.cs [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+        private string GetIcon(CipherView cipher)
+        {
+            string icon = null;
+            switch (cipher.Type)
+            {
+                case CipherType.Login:
+                    icon = GetLoginIconGlyph(cipher);
+                    break;
+                case CipherType.SecureNote:
+                    icon = "";
```
[the remark this answers is not in the document]
Comment (mportune-bw) [reply src/App/Utilities/IconGlyphConverter.cs]: I _think_ I got 'em all...

File: src/App/Controls/CipherViewCell/CipherViewCellViewModel.cs (hunk 1 above)
Comment (mportune-bw) [reply src/App/Controls/CipherViewCell/CipherViewCellViewModel.cs]: Simplified plus added 'http' check (bug discovered while working on #1384 )

File: src/Android/Renderers/ExtendedGridRenderer.cs (hunk 5 above)
Comment (mportune-bw) [reply src/Android/Renderers/ExtendedGridRenderer.cs:30]: The other themes fall back on generic gray levels for light vs dark, whereas nord has specific colors unique to it.

**Review by cscharf — APPROVED** [review]

**Review by cscharf — APPROVED** [review]
Huzzah! Finally get to merge this beast, great work!

**Pull request merged**